sdkproject/Assets/Mapbox/Unity/Editor/PropertyDrawers/VectorLayerPropertiesDrawer.cs

```diff
@@ -116,12 +142,36 @@ public override void OnGUI(Rect position, SerializedProperty property, GUIConten
 					layerSourceId.stringValue = sourcePropertyValue.Id;
 					GUI.enabled = false;
 					EditorGUILayout.PropertyField(sourceOptionsProperty, _mapIdGui);
+					if (_isInitialized)
+					{
+						LoadEditorTileJSON(sourceTypeValue, layerSourceId.stringValue);
+					}
+					else
+					{
+						_isInitialized = true;
+					}
+					if(tileJSONData.PropertyDisplayNames.Count==0)
+					{
+						EditorGUILayout.HelpBox("Invalid Map Id / There might be a problem with the internet connection.", MessageType.Error);
+					}
 					GUI.enabled = true;
 					isActiveProperty.boolValue = true;
 					break;
 				case VectorSourceType.Custom:
 					layerSourceId.stringValue = CustomSourceMapId;
 					EditorGUILayout.PropertyField(sourceOptionsProperty, _mapIdGui);
+					if (_isInitialized)
```

Comment:
Seems like this section is identical to the one just before it; wondering if we can avoid this redundancy somehow